File: lib/antlr-2.7.5/examples/java/calc/Calc.java
import java.io.*;
import antlr.CommonAST;
import antlr.collections.AST;
import antlr.DumpASTVisitor;
import antlr.RecognitionException;
import antlr.TokenStreamException;

class Calc {
	public static void main(String[] args) {
		try {
			CalcLexer lexer = new CalcLexer(new DataInputStream(System.in));
			lexer.setFilename("<stdin>");
			CalcParser parser = new CalcParser(lexer);
			parser.setFilename("<stdin>");
			// Parse the input expression
			parser.expr();
			CommonAST t = (CommonAST)parser.getAST();
			// Print the resulting tree out in LISP notation
			System.out.println(t.toStringTree());
			CalcTreeWalker walker = new CalcTreeWalker();
			// Traverse the tree created by the parser
			float r = walker.expr(t);
			System.out.println("value is "+r);
		}
		catch(TokenStreamException e) {
			System.err.println("exception: "+e);
		}
		catch(RecognitionException e) {
			System.err.println("exception: "+e);
		}
	}
}

File: lib/antlr-2.7.5/examples/csharp/calc/Calc.cs
using System;
//import java.io.*;

using CommonAST				= antlr.CommonAST;
using AST					= antlr.collections.AST;
//using DumpASTVisitor		= antlr.DumpASTVisitor;
using CharBuffer			= antlr.CharBuffer;
using RecognitionException	= antlr.RecognitionException;
using TokenStreamException	= antlr.TokenStreamException;

// wh: bug(?) in DotGNU 0.6 - "using antlr" will workaround the problem.
#if __CSCC__
using antlr;
#endif

class Calc 
{
	public static void Main(string[] args) 
	{
		try 
		{
			CalcLexer lexer = new CalcLexer(new CharBuffer(Console.In));
			lexer.setFilename("<stdin>");
			CalcParser parser = new CalcParser(lexer);
			parser.setFilename("<stdin>");

			// Parse the input expression
			parser.expr();
			CommonAST t = (CommonAST)parser.getAST();
			
			// Print the resulting tree out in LISP notation
			Console.Out.WriteLine(t.ToStringTree());
			CalcTreeWalker walker = new CalcTreeWalker();
			
			// Traverse the tree created by the parser
			float r = walker.expr(t);
			Console.Out.WriteLine("value is "+r);
		}
		catch(TokenStreamException e) 
		{
			Console.Error.WriteLine("exception: "+e);
		}
		catch(RecognitionException e) 
		{
			Console.Error.WriteLine("exception: "+e);
		}
	}  
}
